go-controller/pkg/clustermanager/node/node_allocator.go

```diff
@@ -195,27 +195,24 @@ func (na *NodeAllocator) NeedsNodeAllocation(node *corev1.Node) bool {
 	}
 
 	// ovn node check
-	// allocation is all or nothing, so if one field was allocated from:
-	// nodeSubnets, joinSubnet, layer 2 tunnel id, then all of them were
 	if na.hasNodeSubnetAllocation() {
-		if util.HasNodeHostSubnetAnnotation(node, na.netInfo.GetNetworkName()) {
-			return false
+		if !util.HasNodeHostSubnetAnnotation(node, na.netInfo.GetNetworkName()) {
```

Comment:
just a note, although I agree with your reasoning for changing this logic it might have adverse impact on scale since we are now going to parse all annotations.

Reply:
ack. For now we will just play it safe to avoid regression. In the future we can mitigate our json unmarshal issues with a different CRD/solution.